SimCalorimetry/EcalEBTrigPrimAlgos/src/EcalEBPhase2TrigPrimAlgo.cc

```diff
@@ -0,0 +1,297 @@
+#include <string>
+#include <algorithm>
+#include <numeric>
+#include <functional>
+
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+//#include "Geometry/CaloGeometry/interface/CaloGeometry.h"
+//#include "Geometry/Records/interface/CaloGeometryRecord.h"
+#include "Geometry/EcalMapping/interface/EcalElectronicsMapping.h"
+#include "Geometry/EcalMapping/interface/EcalMappingRcd.h"
+#include "Geometry/EcalAlgo/interface/EcalBarrelGeometry.h"
+
+#include "SimCalorimetry/EcalEBTrigPrimAlgos/interface/EcalEBPhase2TrigPrimAlgo.h"
+
+#include "DataFormats/EcalDetId/interface/EcalSubdetector.h"
+#include "DataFormats/EcalDigi/interface/EBDataFrame_Ph2.h"
+#include "DataFormats/EcalDetId/interface/EcalTrigTowerDetId.h"
+#include "DataFormats/EcalDetId/interface/EBDetId.h"
+#include "DataFormats/EcalDetId/interface/EcalTriggerElectronicsId.h"
+
+#include "CondFormats/EcalObjects/interface/EcalTPGPedestals.h"
+#include "CondFormats/DataRecord/interface/EcalTPGPedestalsRcd.h"
+
+#include "DataFormats/EcalDigi/interface/EcalConstants.h"
+
+#include <TTree.h>
+#include <TMath.h>
+
+//----------------------------------------------------------------------
+
+const unsigned int EcalEBPhase2TrigPrimAlgo::nrSamples_ = ecalPh2::sampleSize;  // 16 samples
+const unsigned int EcalEBPhase2TrigPrimAlgo::maxNrTowers_ = 2448;               // number of towers in EB
+
+EcalEBPhase2TrigPrimAlgo::EcalEBPhase2TrigPrimAlgo(const EcalTrigTowerConstituentsMap *eTTmap,
+                                                   const CaloGeometry *theGeometry,
+                                                   int binofmax,
+                                                   bool debug)
+    : eTTmap_(eTTmap),
+      theGeometry_(theGeometry),
+      binOfMaximum_(binofmax),
+      debug_(debug)
+
+{
+  maxNrSamples_ = ecalPh2::sampleSize;
+  this->init();
+}
+
+void EcalEBPhase2TrigPrimAlgo::init() {
+  theMapping_ = new EcalElectronicsMapping();
+  // initialise data structures
+  initStructures(towerMapEB_);
+  hitTowers_.resize(maxNrTowers_);
+
+  linearizer_ = new EcalEBPhase2Linearizer(debug_);
+  lin_out_.resize(maxNrSamples_);
+
+  amplitude_reconstructor_ = new EcalEBPhase2AmplitudeReconstructor(debug_);
+  filt_out_.resize(maxNrSamples_);
+
+  tpFormatter_ = new EcalEBPhase2TPFormatter(debug_);
```

Comment:
Please delete this object when it is not needed anymore.

Reply:
Done. Had gone unnoticed